tests/check_mlflow_lazily_imports_ml_packages.py

```diff
@@ -0,0 +1,54 @@
+"""
+Tests that `import mlflow` and `mlflow.autolog()` do not import ML packages.
+"""
+
+import sys
+import importlib
+import mlflow  # pylint: disable=unused-import
+
+
+def main():
+    ml_packages = {
+        "catboost",
+        "fastai",
+        "mxnet",
+        "h2o",
+        "keras",
+        "lightgbm",
+        "mleap",
+        "onnx",
+        "pytorch_lightning",
+        "pyspark",
+        "shap",
+        "sklearn",
+        "spacy",
+        "statsmodels",
+        "tensorflow",
+        "torch",
+        "xgboost",
+        "pmdarima",
+        "diviner",
+    }
+    imported = ml_packages.intersection(set(sys.modules))
+    assert imported == set(), f"mlflow imports {imported} when it's imported but it should not"
+
+    mlflow.autolog()
+    imoprted = ml_packages.intersection(set(sys.modules)) - {"pyspark"}
```

Comment:
You might need to exclude "pyspark.ml" as well.

Reply:
For now, we don't need to.

Reply:
Added `pyspark.ml`